_query-dsl/term/term.md

```diff
@@ -42,6 +42,9 @@ GET shakespeare/_search
 ```
 {% include copy-curl.html %}
 
+In OpenSearch 2.x and previous versions, every (alphabetic) character in such a query doubles the complexity of the search, consuming heap memory and potentially crashing nodes due to high CPU with GC thrashing. Even a relatively short search term (about 16 characters) could result in nearly 8 GB of heap. Case-insensitive searches should be avoided by using a [lowercase token filter]({{site.url}}{{site.baseurl}}/analyzers/token-filters/lowercase/) on the indexed field's analyzer, and then using lowercase query terms instead of relying on case-insensitive searches.
```

Comment:
I would be intentionally a little more vague. I don't know the details of Hopcroft's algorithm to definitively say that it's 2x per character. (If you know and are sure, then please ignore me!)

I think we can be hand-wavy and say that "complexity may scale exponentially with the number of characters". That's probably scary enough to keep folks safe.

The recommendation to solve the problem at indexing time is (imo) the best advice.